### ISSUE TITLE
Unable to update a child instance with a composite PK

### ISSUE THREAD

**Issue by blaluc** [agrestio/agrest#370]
If I try to save/update an object with a child entity having a composite PK I get this error:
```
java.lang.IllegalArgumentException: Entity 'MyChildEntity' has unexpected number of ID attributes: 2
	at io.agrest.runtime.protocol.EntityUpdateJsonTraverser.converter(EntityUpdateJsonTraverser.java:177) ~[agrest-3.0.jar:na]
....
```

How to deal with this kind of issue, considering that the database is already in place and I cannot change its structure by replacing composite keys with single columns PK?

**Comment by andrus** [comment]
What is the semantics of this relationship at the DB level. Could you show the PK/FK structure of the parent and child tables?

**Comment by scali** [comment]
Same issue for me with this semantics (PostgreSQL database) and Agrest 4.6-SNAPSHOT

```
CREATE TABLE public.A
(
    id serial,
    name character varying NOT NULL
}

CREATE TABLE public.B
(
    id serial,
    name character varying NOT NULL
}

CREATE TABLE public.partnership
(
    a integer NOT NULL,
    b integer NOT NULL,
    partnership_type character varying NOT NULL,
    CONSTRAINT pk_partnership PRIMARY KEY (a, b),
    CONSTRAINT fk_partnership_a FOREIGN KEY (a)
            REFERENCES public.a (id) MATCH SIMPLE
            ON UPDATE NO ACTION
            ON DELETE NO ACTION,
    CONSTRAINT fk_partnership_b FOREIGN KEY (b)
            REFERENCES public.b (id) MATCH SIMPLE
            ON UPDATE NO ACTION
            ON DELETE NO ACTION
);
```

**Comment by scali** [comment]
It's seems finally not to be a supported feature yet, as [this part of code](https://github.com/agrestio/agrest/blob/master/agrest-engine/src/main/java/io/agrest/runtime/protocol/EntityUpdateJsonTraverser.java#L156) (from the class `agrest/agrest-engine/src/main/java/io/agrest/runtime/protocol/EntityUpdateJsonTraverser.java`)  will always throw an exception for composite PK :

```
 private JsonValueConverter<?> converter(AgRelationship relationship) {
        AgEntity<?> target = relationship.getTargetEntity();
        int ids = target.getIdParts().size();
        if (ids != 1) {
            throw new IllegalArgumentException("Entity '" + target.getName() +
                    "' has unexpected number of ID attributes: " + ids);
        }
        return converterFactory.converter(target.getIdParts().iterator().next().getType());
    }
```

**Comment by andrus** [comment]
@scali : yeah, this is the case here. This limitation is somewhat arbitrary, and we need to address it. 

In your case however addressing the above will not help directly, as there is no semantics in Agrest to *create* related objects. So something like `{"name":"a1","partnerships":[{"a":1,"b":2, "partnershipType":"x"},...]}`, will try to look for an existing partnership. 

What we really need is a way to specify creation of nested object trees (graphs?) in one shot (something that would be very useful in many cases, say create an Account with Addresses). This will require Agrest to understand somehow by looking at JSON which subobject in a graph is expected to be present in DB, and which needs to be created, and which - merged on top of the existing one. Request method (POST, PUT) applies only to the root entity and is not sufficient to make those decisions once you traverse the graph to root's children.  Something I'd very much like to discuss and solve.

In the meantime, the workaround for your case would be to split the update into 2 endpoints (one - for A (or B), and one - for Partnership).
